### PR TITLE
SQ-57/venue screen

### PR DESCRIPTION
**Pull request by danybony**
This PR adds a MVP version of the venue screen.
It already displays:
* name
* address
* description
* map static image

All the data come from the Firebase Venue node.

#### Still missing
* implement coordinator layout scrolling behaviour
* implement navigate to Google Maps clicking on the map

### Screenshot

device-2017-03-22-124714

### CODE REVIEW

File: app/src/main/java/net/squanchy/service/repository/injection/RepositoryModule.java [hunk 1]
```diff
@@ -27,4 +28,11 @@ SpeakerRepository speakerRepository(
     ) {
         return new SpeakerRepository(dbService, checksum);
     }
+
+    @Provides
+    VenueRepository venueRepositoryrRepository(
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/service/repository/injection/RepositoryModule.java]: On one line please

File: app/src/main/java/net/squanchy/venue/VenueInfoComponent.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import net.squanchy.navigation.NavigationModule;
+import net.squanchy.navigation.Navigator;
+
+import dagger.Component;
+
+@ActivityLifecycle
+@Component(modules = {VenueInfoModule.class, NavigationModule.class}, dependencies = ApplicationComponent.class)
+interface VenueInfoComponent {
+
+    Context context();
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/venue/VenueInfoComponent.java]: Why do you need this here? Are you injecting the `Context` in the view? Or is there something I'm missing?

File: app/src/main/java/net/squanchy/venue/VenueInfoModule.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import net.squanchy.service.repository.VenueRepository;
+
+import dagger.Module;
+import dagger.Provides;
+
+@Module
+class VenueInfoModule {
+
+    @Provides
+    VenueInfoService favoritesService(
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/venue/VenueInfoModule.java]: On one line please

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    public VenueInfoPageView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr, defStyleRes);
+
+        super.setOrientation(VERTICAL);
+    }
+
+    @Override
+    public void setOrientation(int orientation) {
+        throw new UnsupportedOperationException("TweetsPageView doesn't support changing orientation");
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.java]: Copypasta error in the message

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+        nameText = (TextView) findViewById(R.id.venue_name);
+        addressText = (TextView) findViewById(R.id.venue_address);
+        descriptionText = (TextView) findViewById(R.id.venue_description);
+        mapView = (ImageView) findViewById(R.id.venue_map);
+
+        setupToolbar();
+    }
+
+    private static Activity unwrapToActivityContext(Context context) {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.java]: I suppose we could extract this if we use it in several places now, to a helper class or something.

File: app/src/main/java/net/squanchy/venue/VenueInfoService.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+class VenueInfoService {
+
+    private final VenueRepository venueRepository;
+
+    VenueInfoService(VenueRepository venueRepository) {
+        this.venueRepository = venueRepository;
+    }
+
+    public Observable<Venue> venue() {
+        return venueRepository.venue();
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/venue/VenueInfoService.java]: ❌ need to ensure the user is logged in (as anon user at least). See how other services do it.

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -252,4 +252,56 @@
     <item name="fontPath">?defaultMediumTypeface</item>
   </style>
 
+  <style name="VenueInfo.Label" parent="None">
+    <item name="android:gravity">center_vertical</item>
```
Comment (rock3r) [inline app/src/main/res/values/styles.xml]: Not 100% necessary but, for clarity, put the horizontal gravity too: `start|center_vertical`

**Review by rock3r — CHANGES_REQUESTED** [review]
One change required and a few nice-to-haves, but nice one! 💯

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.java (hunk 4 above)
Comment (danybony) [reply app/src/main/java/net/squanchy/venue/VenueInfoPageView.java]: Ooops 👅

File: app/src/main/java/net/squanchy/venue/VenueInfoService.java (hunk 6 above)
Comment (danybony) [reply app/src/main/java/net/squanchy/venue/VenueInfoService.java]: Ah right, good point! I'll have a look

**Review by rock3r — APPROVED** [review]
Nice!

**Pull request merged**